language/stdlib/modules/DiemAccount.move

```diff
@@ -1489,6 +1492,31 @@ module DiemAccount {
         aborts_if !DiemTransactionPublishingOption::spec_is_module_allowed(sender) with Errors::INVALID_STATE;
     }
 
+    /// If published, halts transactions from all accounts except DiemRoot
+    resource struct HaltAllTransactions {}
+
+    /// If called, transactions cannot be sent from any account except DiemRoot
+    public fun halt_all_transactions(dr_account: &signer) {
+        Roles::assert_diem_root(dr_account);
+        assert(
+            !exists<HaltAllTransactions>(Signer::address_of(dr_account)),
+            Errors::already_published(EHALT_ALL_TRANSACTIONS),
+        );
+        move_to(dr_account, HaltAllTransactions {})
+    }
+
+    /// If called, transactions can be sent from any account once again
+    public fun resume_transactions(dr_account: &signer) acquires HaltAllTransactions {
+        Roles::assert_diem_root(dr_account);
+        let dr_address = Signer::address_of(dr_account);
+        assert(
+            exists<HaltAllTransactions>(dr_address),
+            Errors::already_published(EHALT_ALL_TRANSACTIONS),
+        );
+
+        let HaltAllTransactions {} = move_from<HaltAllTransactions>(dr_address);
+    }
```

Comment:
It feels like this logic would be better placed/relevant in `DiemTransactionPublishingOption`  to me. Or maybe there was a particular reason to add it here?

Reply:
I added here because we need to look at `HaltAllTransactions` in the prologue. We could move this to `DiemTransactionPublishingOption` instead, but then we'd need to expose a public `exists` wrapper in addition to these two functions. But perhaps seems worth doing for consistency with `is_script_allowed`.

Reply:
Fixed.